pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -257,3 +258,28 @@ func validateResources(resources []config.ResourceSpec) error {
 	}
 	return nil
 }
+
+// ValidateNodeAffinityArgs validates that NodeAffinityArgs are correct.
+func ValidateNodeAffinityArgs(args *config.NodeAffinityArgs) error {
+	if args.AddedAffinity == nil {
+		return nil
+	}
+	affinity := args.AddedAffinity
+	f := field.NewPath("addedAffinity")
+	var allErrs field.ErrorList
+	if ns := affinity.RequiredDuringSchedulingIgnoredDuringExecution; ns != nil {
+		_, err := nodeaffinity.NewNodeSelector(ns)
+		if err != nil {
+			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
+			allErrs = append(allErrs, field.Invalid(f.Child("requiredDuringSchedulingIgnoredDuringExecution"), affinity.RequiredDuringSchedulingIgnoredDuringExecution, err.Error()))
+		}
+	}
+	if terms := affinity.PreferredDuringSchedulingIgnoredDuringExecution; len(terms) != 0 {
+		_, err := nodeaffinity.NewPreferredSchedulingTerms(terms)
+		if err != nil {
+			// TODO(#96167): Expand all field.Error(s) returned once constructor use them.
+			allErrs = append(allErrs, field.Invalid(f.Child("preferredDuringSchedulingIgnoredDuringExecution"), affinity.PreferredDuringSchedulingIgnoredDuringExecution, err.Error()))
+		}
+	}
+	return allErrs.ToAggregate()
```

Comment:
add a TODO to validate RequiredDuringSchedulingRequiredDuringExecution once implemented

Reply:
Done